nbclient/client.py

```diff
@@ -223,6 +223,35 @@ class NotebookClient(LoggingConfigurable):
 
     kernel_manager_class = Type(config=True, help='The kernel manager class to use.')
 
+    on_kernel_create = Any(
+        default_value=None,
+        allow_none=True,
+        help="""A callable which executes when the kernel is created.""",
+    ).tag(config=True)
+
+    on_cell_start = Any(
```

Comment:
These are somewhat awkward as `Any` traitlets :/ Until we decide to move off them I guess this is how it'd be

Reply:
agreed...

Reply:
Should we use a `Callable` trait? It should also be typed as `t.Callable`.

Reply:
Is there a Callable trait? I couldn't find one, and I copied the Any from the timeout func.

IMO, typing on traits is a bit redundant

Reply:
You're right, it looks like `Callable` has been added to `traitlets` in the past but then removed. I still think static typing traits is valuable, because it can catch bugs before runtime. You're right, it should be `t.Optional[t.Callable]`.